### DISCUSSION TITLE
Thoughts for v5

### DISCUSSION THREAD

**Discussion by imbhargav5** [imbhargav5/rooks#406] [Ideas]
Hi Community,

I have been thinking about some improvements to Rooks for a while and I wanted to share with you all. I think for us to become a more viable alternative, we need to not only be strong technically with a solid collection of hooks, but we should also be a breeze to use for developers.

We launched some very handy tools for devs in v4. 
- Move to Typescript
- Enhanced autocomplete which is type safe
- Added documentation via docusaurus and more

For v5, we need to get more things in place to be a powerful solution . Here are my first few items in the list.
- An automatic changelog
- Simplify rooks by only publishing one package (`rooks`) All independent packages can be used from `rooks` anyway and they don't add to bundle size as they are fully ESM compliant.
- Clean up repo to be a simple one package repo and not a monorepo.
- Move to ESM build ?? 🤔 

Let me know what you think! Also cc @maciekgrzybek, @simbathesailor  your thoughts? Everyone is welcome to join this discussion as always!

**Comment by maciekgrzybek** [comment]
Yeah that's all sounds really good :) We could try using https://github.com/evanw/esbuild? Apparently is much much faster. I also think it would be good to add unit or integration tests for all the hooks, I think this is something that's missing.

**Reply by imbhargav5** [reply]
Absolutely. will add this to the list as well. Integration tests probably won't make sense for hooks. We already have unit tests for a lot of our hooks but we should improve coverage as you brought up!
